modules/ztp-image-based-upgrade.adoc

```diff
@@ -0,0 +1,71 @@
+
+[IMPORTANT]
+====
+The image-based upgrade uses custom images that are specific to the hardware platform that the clusters are running on.
+Each different hardware platform requires a separate custom image.
+====
+// TODO agree on how much we should specify this. Is this enough or detail about CPU topology, deployment method, etc.
+
+The {lcao} uses two custom resources (CRs) on the participating clusters:
+
+* On the seed cluster, the `SeedGenerator` CR allows the seed image generation. This CR specifies the repository to push the seed image to.
+* On the target cluster, the `ImageBasedUpgrade` CR specifies the seed container image for the upgrade of the target cluster and the backup configurations for your workloads.
+
+After generating the seed image on the seed cluster, you can move throught the stages on the target cluster by setting the `spec.stage` field to the following values in the `ImageBasedUpgrade` CR:
+
+* `Idle`
+* `Prep`
+* `Upgrade`
+* `Rollback` (Optional)
+* `Idle`
+
+[discrete]
+== Idle stage
+
+This is the default stage when the {lcao} is installed. There is no ongoing upgrade and the cluster is ready to move to the `Prep` stage.
+
+After a successful upgrade or a rollback, you commit to the change by manually moving the stage to `Idle`.
+Changing to this stage ensures that the {lcao} cleans up resources, so the cluster is ready for upgrades.
+
+[discrete]
+== Prep stage
+
+You can complete this stage before a scheduled maintenance window.
+
+In the `Prep` stage, the {lcao} ensures that the target clusters are ready to proceed to the `Upgrade` stage by checking if they meet certain conditions.
+The Operator uses the `ostree` versioning system and the OADP Operator to create a backup, which allow complete cluster reconfiguration after upgrade or rollback.
+
+At the `Prep` stage, the {lcao} pulls the seed image and additional container images specified in the seed image.
+
+You can stop the upgrade process. If you stop, the Operator performs cleanup operations and sets the upgrade stage back to `Idle`.
+
+[discrete]
+== Upgrade stage
+
+Just before the {lcao} starts the upgrade process, the OADP Operator creates a backup of the user-specified resources on a compatible Object storage solution, such as an Amazon S3 bucket.
+These OADP `Backup` and `Restore` CRs are applied to the target cluster after the upgrade.
+Once the upgrade is successful, the target cluster boots with the new platform version, then the Operator applies the cluster and application configurations.
+
+The Operator also regenerates the cluster certificate.
+This ensures that each {sno} cluster upgraded with the same seed image has unique and valid cryptographic objects.
+
+[discrete]
+== Rollback stage (Optional)
+
+You can use the backup from the `Upgrade` stage to manually start a rollback if the upgrade fails.
```

Comment:
Maybe it is confusing by using the word backup. Do you mean the backup created by OADP or the previous stateroot? For rollback it just boots in the previous stateroot, basically setting it as the default boot.

Reply:
I meant the OADP backup but now I realize that I still confuse when backup vs previous stateroot is used.
To avoid mentioning stateroots, I changed it like this:
> You can manually start a rollback if the upgrade fails.
> If you roll back the upgrade, the cluster boots up with the previous release of {product-title} and application configurations.